lib/phoenix/live_dashboard/components/shared_usage_card_component.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Phoenix.LiveDashboard.SharedUsageCardComponent do
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  alias Phoenix.LiveDashboard.{
+    ColorBarComponent,
+    ColorBarLegendComponent
+  }
+
+  @csp_nonces %{img: nil, script: nil, style: nil}
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  def normalize_params(params) do
+    params
+    |> validate_required([:usages, :total_data, :total_legend, :total_usage, :dom_id])
+    |> validate_usages()
+    |> put_defaults()
+  end
+
+  defp validate_usages(params = %{usages: usages}) do
+    normalized_usages =
+      Enum.map(usages, fn usage ->
+        validate_required(usage, [:data, :dom_sub_id])
+        put_usage_defaults(usage)
+      end)
+
+    %{params | usages: normalized_usages}
+  end
+
+  defp validate_required(params, list) do
+    case Enum.find(list, &(not Map.has_key?(params, &1))) do
+      nil -> :ok
+      key -> raise ArgumentError, "expected #{inspect(key)} parameter to be received"
+    end
+
+    params
+  end
+
+  defp put_usage_defaults(params) do
+    params
+    |> Map.put_new(:title, nil)
+  end
+
+  defp put_defaults(params) do
+    params
+    |> Map.put_new(:title, nil)
+    |> Map.put_new(:hint, nil)
+    |> Map.put_new(:inner_title, nil)
+    |> Map.put_new(:inner_hint, nil)
+    |> Map.put_new(:csp_nonces, @csp_nonces)
+    |> Map.put_new(:total_formatter, &"#{&1} %")
+  end
+
+  @impl true
+  def render(assigns) do
+    ~L"""
+    <%= if @title do %>
+      <h5 class="card-title">
+        <%= @title %>
+        <%= if @hint do %>
+          <%= hint(do: @hint) %>
+        <% end %>
+      </h5>
+    <% end %>
+    <div class="card">
+      <%= if @inner_title do %>
+        <h5 class="card-title">
+          <%= @inner_title %>
+          <%= if @inner_hint do %>
+            <%= hint(do: @inner_hint) %>
+          <% end %>
+        </h5>
+      <% end %>
+      <div class="card-body">
+        <div phx-hook="PhxColorBarHighlight" id="cpu-color-bars">
+          <%= for usage <- @usages do %>
+            <div class="flex-grow-1 mb-3">
+              <%= live_component @socket, ColorBarComponent, dom_id: "cpu-#{usage.dom_sub_id}", data: usage.data, title: usage.title, csp_nonces: @csp_nonces %>
+            </div>
+          <% end %>
+          <%= live_component @socket, ColorBarLegendComponent, data: @total_data, formatter: @total_formatter %>
```

Comment:
It would be nice to remove this component and instead make if a function or just inline the code directly here. :) The same for ColorBarComponent above.

Reply:
I removed the components and added the code directly to the render.